src/Akeneo/Pim/Structure/Bundle/Resources/public/js/attribute-option/components/AttributeOptions.tsx

```diff
@@ -1,58 +1,79 @@
 import React, {useEffect, useState} from 'react';
 import List from './List';
 import Edit from './Edit';
+import New from './New';
 import {useDispatch, useSelector} from 'react-redux';
 import {AttributeOptionsState} from '../store/store';
 import {AttributeOption} from '../model';
 import {useSaveAttributeOption} from '../hooks/useSaveAttributeOption';
-import {updateAttributeOptionAction} from '../reducers';
+import {useCreateAttributeOption} from '../hooks/useCreateAttributeOption';
+import {updateAttributeOptionAction, createAttributeOptionAction} from '../reducers';
+import {useAttributeContext} from '../contexts';
 
 const AttributeOptions = () => {
     const attributeOptions = useSelector((state: AttributeOptionsState) => state.attributeOptions);
     const [selectedOption, setSelectedOption] = useState<AttributeOption | null>(null);
     const [isSaving, setIsSaving] = useState<boolean>(false);
+    const [showNewOptionForm, setShowNewOptionForm] = useState<boolean>(false);
     const attributeOptionSaver = useSaveAttributeOption();
+    const attributeOptionCreate = useCreateAttributeOption();
     const dispatchAction = useDispatch();
+    const attribute = useAttributeContext();
 
     useEffect(() => {
-        if (attributeOptions !== null && attributeOptions.length > 0) {
+        if (attributeOptions !== null && attributeOptions.length > 0 && selectedOption === null) {
             setSelectedOption(attributeOptions[0]);
-        } else {
-            setSelectedOption(null);
         }
     }, [attributeOptions]);
 
     useEffect(() => {
         setSelectedOption(null);
+    }, [attribute.attributeId]);
 
-        return () => {
-            setSelectedOption(null);
-        };
-    }, []);
-
-    const onSelectAttributeOption = (optionId: number) => {
+    const selectAttributeOption = (optionId: number | null) => {
         if (attributeOptions !== null) {
             const option = attributeOptions.find((option: AttributeOption) => option.id === optionId);
             if (option !== undefined) {
                 setSelectedOption(option);
+                setShowNewOptionForm(false);
+
+                return;
             }
         }
+
+        setShowNewOptionForm(false);
+        setSelectedOption(null);
     };
 
-    const onSaveAttributeOption = async (attributeOption: AttributeOption) => {
+    const saveAttributeOption = async (attributeOption: AttributeOption) => {
         setIsSaving(true);
         await attributeOptionSaver(attributeOption);
         setIsSaving(false);
         dispatchAction(updateAttributeOptionAction(attributeOption));
     };
 
+    const createAttributeOption = async (optionCode: string) => {
+        setIsSaving(true);
+        const attributeOption = await attributeOptionCreate(optionCode);
+        setIsSaving(false);
+        dispatchAction(createAttributeOptionAction(attributeOption));
+        setShowNewOptionForm(false);
+        setSelectedOption(attributeOption);
+    };
+
```

Comment:
I think we should encapsulate this method in a useCallback hook to avoid multiple re rendering because it is used as a props in the New component below.

```suggestion
    const createAttributeOption = useCallback(async (optionCode: string) => {
        setIsSaving(true);
        const attributeOption = await attributeOptionCreate(optionCode);
        setIsSaving(false);
        dispatchAction(createAttributeOptionAction(attributeOption));
        setShowNewOptionForm(false);
        setSelectedOption(attributeOption);
    }, [attributeOptionCreate, setIsSaving, dispatchAction, setShowNewOptionForm, setSelectedOption]);

```